### ISSUE TITLE
Update libcrypto3/libssl3 to fix CVE-2023-2650

### ISSUE THREAD

**Issue by jasonrandersen00** [alpinelinux/docker-alpine#328]
Hi and thanks for maintaining this!
I hope this isn't a duplicate. 
Today our vulnerability scanner failed because of the libcrypto3/libssl3 CVE-2023-2650 vulnerability.

```
NAME        INSTALLED  FIXED-IN  TYPE  VULNERABILITY  SEVERITY 
libcrypto3  3.1.0-r4   3.1.1-r0  apk   CVE-2023-2650  High      
libssl3     3.1.0-r4   3.1.1-r0  apk   CVE-2023-2650  High  
```

Running `apk upgrade libssl3 libcrypto3` in my Dockerfile manually fixes the issue.

It would be awesome if you could update the base image to include the latest fixes :)

**Comment by melchisedech333** [comment]
Thanks a lot, this worked and I was able to get my pipelines through the security tests.

**Comment by shvkumara** [comment]
This is not working for us, we are not able to upgrade the packages to the fixed ones.
```
% docker run -it alpine:3.18.0 sh
/ # apk list | grep libcrypto3
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/main: No such file or directory
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/community: No such file or directory
libcrypto3-3.1.0-r4 aarch64 {openssl} (Apache-2.0) [installed]
/ # apk list | grep libssl
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/main: No such file or directory
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/community: No such file or directory
libssl3-3.1.0-r4 aarch64 {openssl} (Apache-2.0) [installed]
/ # apk upgrade libssl3 libcrypto3
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.18/community/aarch64/APKINDEX.tar.gz
OK: 8 MiB in 15 packages
/ # apk list | grep libcrypto3
libcrypto3-3.1.0-r4 aarch64 {openssl} (Apache-2.0) [installed]
/ # apk list | grep libssl
libressl3.7-libssl-3.7.3-r0 aarch64 {libressl} (custom)
libssl1.1-1.1.1t-r2 aarch64 {openssl1.1-compat} (OpenSSL)
libssl3-3.1.0-r4 aarch64 {openssl} (Apache-2.0) [installed]
```

**Comment by shvkumara** [comment]
```
% cat /tmp/Dockerfile 
FROM alpine:3.18.0
RUN apk upgrade libssl3 libcrypto3

% docker build -t test:1 /tmp     
[+] Building 5.9s (6/6) FINISHED                                                                                                                                      
 => [internal] load build definition from Dockerfile                                                                                                             0.0s
 => => transferring dockerfile: 91B                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/alpine:3.18.0                                                                                                 0.0s
 => CACHED [1/2] FROM docker.io/library/alpine:3.18.0                                                                                                            0.0s
 => [2/2] RUN apk upgrade libssl3 libcrypto3                                                                                                                     5.9s
 => exporting to image                                                                                                                                           0.0s
 => => exporting layers                                                                                                                                          0.0s
 => => writing image sha256:4569a8f46b4f9ee3973403a4b47ae08c95dd7841e2e7243be2a1ed3f7f90f0db                                                                     0.0s
 => => naming to docker.io/library/test:1                                                                                                                        0.0s

% trivy image test:1
2023-06-08T15:36:45.977+0530    INFO    Vulnerability scanning is enabled
2023-06-08T15:36:45.977+0530    INFO    Secret scanning is enabled
2023-06-08T15:36:45.977+0530    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-06-08T15:36:45.977+0530    INFO    Please see also https://aquasecurity.github.io/trivy/v0.41/docs/secret/scanning/#recommendation for faster secret detection
2023-06-08T15:36:46.165+0530    INFO    Detected OS: alpine
2023-06-08T15:36:46.166+0530    INFO    This OS version is not on the EOL list: alpine 3.18
2023-06-08T15:36:46.166+0530    INFO    Detecting Alpine vulnerabilities...
2023-06-08T15:36:46.168+0530    INFO    Number of language-specific files: 0

test:1 (alpine 3.18.0)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 2, CRITICAL: 0)

┌────────────┬───────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────┐
│  Library   │ Vulnerability │ Severity │ Installed Version │ Fixed Version │                       Title                       │
├────────────┼───────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────┤
│ libcrypto3 │ CVE-2023-2650 │ HIGH     │ 3.1.0-r4          │ 3.1.1-r0      │ Possible DoS translating ASN.1 object identifiers │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-2650         │
├────────────┤               │          │                   │               │                                                   │
│ libssl3    │               │          │                   │               │                                                   │
│            │               │          │                   │               │                                                   │
└────────────┴───────────────┴──────────┴───────────────────┴───────────────┴───────────────────────────────────────────────────┘
```

**Comment by arwinters** [comment]
It did work for me today. Can you post your Dockerfile

**Comment by PriyaKumar2203** [comment]
I am using eclipse-temurin:17.0.6_10-jdk-alpine docker image, Spring boot 3.0.6 and Java 17 in our project.

Can you please mention the docker image as well?

**Comment by codeaprendiz** [comment]
The reason why it's not working for some is because the patched packages have been released for `Alpine Main x86_64` and not for `Alpine Main aarch64`.

- [libcrypt3](https://pkgs.org/download/libcrypto3)
- [libssl3](https://pkgs.org/download/libssl3)

Here's a simple test to validate

- Dockerfile

```Dockerfile
FROM alpine:3.18.0
RUN apk update && apk add --upgrade libcrypto3 libssl3
```

## Build on aarch64

```bash
❯ cat Dockerfile                       
FROM alpine:3.18.0
RUN apk update && apk add --upgrade libcrypto3 libssl3

❯ docker build -t alpine:local  .    
.                  ### Output truncated

❯ docker run --rm -it alpine:local sh
/ # uname -m
aarch64

## And therefore the latest packages are different
/ # apk info libssl3 libcrypto3 | egrep webpage
libcrypto3-3.1.0-r4 webpage:
libssl3-3.1.0-r4 webpage:
```

- Therefor `trivy` scan fails

```bash
# Run on mac m1
❯ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/Library/Caches:/root/.cache/  aquasec/trivy:0.42.0  image alpine:local

alpine:local (alpine 3.18.0)
============================
Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 2, CRITICAL: 0)

┌────────────┬───────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────┐
│  Library   │ Vulnerability │ Severity │ Installed Version │ Fixed Version │                       Title                       │
├────────────┼───────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────┤
│ libcrypto3 │ CVE-2023-2650 │ HIGH     │ 3.1.0-r4          │ 3.1.1-r0      │ Possible DoS translating ASN.1 object identifiers │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-2650         │
├────────────┤               │          │                   │               │                                                   │
│ libssl3    │               │          │                   │               │                                                   │
│            │               │          │                   │               │                                                   │

```

## Build on x86_64

```bash
# uname -m
x86_64

# cat Dockerfile 
FROM alpine:3.18.0
RUN apk update && apk add --upgrade libcrypto3 libssl3

# docker build -t alpine:local  .
.                  ### Output truncated

# docker run -it alpine:local sh
/ # uname -m
x86_64
# ## And therefore the latest packages are different
/ # apk info libssl3 libcrypto3 | egrep webpage
libcrypto3-3.1.1-r1 webpage:
libssl3-3.1.1-r1 webpage:
```

- So `trivy` can succeeds

```bash
# docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/Library/Caches:/root/.cache/  aquasec/trivy:0.42.0  image alpine:local
.
alpine:local (alpine 3.18.0)
============================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by shvkumara** [comment]
Yes.. building on top of amd64 worked
```
 docker buildx build --platform linux/amd64 -t helm-exporter:1 .
```
Trivy scan
```
helm-exporter:1 (alpine 3.18.0)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Comment by andrealmar** [comment]
We are facing the same issue here. Our pipelines are getting stopped on `trivy` scan step because of this: 

```shell
┌────────────┬───────────────┬──────────┬───────────────────┬───────────────┬───────────────────────────────────────────────────┐
│  Library   │ Vulnerability │ Severity │ Installed Version │ Fixed Version │                       Title                       │
├────────────┼───────────────┼──────────┼───────────────────┼───────────────┼───────────────────────────────────────────────────┤
│ libcrypto3 │ CVE-2023-2650 │ HIGH     │ 3.1.0-r4          │ 3.1.1-r0      │ Possible DoS translating ASN.1 object identifiers │
│            │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-2650         │
├────────────┤               │          │                   │               │                                                   │
│ libssl3    │               │          │                   │               │                                                   │
│            │               │          │                   │               │                                                   │
```

**Comment by aaleksandrov** [comment]
Having the same issue for arm64
```
FROM node:18-alpine3.18
.....
RUN apk update && apk upgrade && apk add --upgrade libcrypto3 libssl3
```
...
Security scanner results:
```
found libcrypto3 3.1.0-r4 CVE-2023-2650 (fixed: 3.1.1-r0)
found libssl3    3.1.0-r4 CVE-2023-2650 (fixed: 3.1.1-r0)
```

**Comment by shvkumara** [comment]
@andrealmar and @aaleksandrov   You need to build the docker image on x86_64, if you are using the mac, then I have pasted the command above on how to build docker image using x86_64.

**Comment by aaleksandrov** [comment]
@shvkumara I am building for Graviton 2
Anyways patches are finally published for all archs so the issue is resolved and can be closed.
https://pkgs.org/download/libcrypto3

**Comment by ncopa** [comment]
CVE-2023-2650 was fixed in[ 3.18.2](https://alpinelinux.org/posts/Alpine-3.15.9-3.16.6-3.17.4-3.18.2-released.html).
